friends/src/App.js

```diff
@@ -0,0 +1,118 @@
+import React, {Component} from 'react';
+import './App.css';
+import axios from 'axios';
+import {Route} from 'react-router-dom';
+import Home from './Home';
+
+export default class App extends Component {
+  constructor(){
+    super();
+    this.state = {
+      data: [],
+      id: 0,
+      name: '',
+      age: 0,
+      email: '',
+      addFriend: false,
+    }
+  }
+
+  componentDidMount = () => {
+    axios.get('http://localhost:5000/friends')
+      .then( response => {
+        if(typeof response.data.message === 'string'){
+          Promise.reject("Error: Friends not found :( ")
+        }
+        this.setState({ data: response.data})
+      })
+      .catch( err=> console.log(err))
+    }
+
+  handleChange = (event)=>{
+    this.setState({
+      [event.target.id]: event.target.value
+    })
+  }
+
+  preventDefault = (e) => {
+    e.preventDefault();
```

Comment:
This function should not be necessary as it is already a method on the event object. For more information, please read: https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault